stdlib/ballerina-http/src/main/java/org/ballerinalang/net/http/WebSocketUtil.java

```diff
@@ -139,52 +134,52 @@ public static void executeOnOpenResource(Resource onOpenResource, BStruct webSoc
+                    readFirstFrame(webSocketConnection, webSocketConnector);
                 }
                 ErrorHandlerUtils.printError("error: " + BLangVMErrors.getPrintableStackTrace(error));
             }
         };
-
         //TODO handle BallerinaConnectorException
-        Executor.submit(onOpenResource, onOpenCallableUnitCallback,
-                        null, null, bValues);
+        Executor.submit(onOpenResource, onOpenCallableUnitCallback, null, null, bValues);
     }
 
     public static void populateEndpoint(WebSocketConnection webSocketConnection, BStruct webSocketEndpoint) {
-        webSocketEndpoint.setStringField(0, webSocketConnection.getId());
-        webSocketEndpoint.setStringField(1, webSocketConnection.getSession().getNegotiatedSubprotocol());
-        webSocketEndpoint.setBooleanField(0, webSocketConnection.getSession().isSecure() ? 1 : 0);
-        webSocketEndpoint.setBooleanField(1, webSocketConnection.getSession().isOpen() ? 1 : 0);
+        webSocketEndpoint.setStringField(WebSocketConstants.LISTENER_ID_INDEX, webSocketConnection.getId());
+        webSocketEndpoint.setStringField(WebSocketConstants.LISTENER_NEGOTIATED_SUBPROTOCOLS_INDEX,
+                                         webSocketConnection.getSession().getNegotiatedSubprotocol());
+        webSocketEndpoint.setBooleanField(WebSocketConstants.LISTENER_IS_SECURE_INDEX,
+                                          webSocketConnection.getSession().isSecure() ? 1 : 0);
+        webSocketEndpoint.setBooleanField(WebSocketConstants.LISTENER_IS_OPEN_INDEX,
+                                          webSocketConnection.getSession().isOpen() ? 1 : 0);
     }
 
     public static void handleWebSocketCallback(Context context, CallableUnitCallback callback,
                                                ChannelFuture webSocketChannelFuture) {
-        webSocketChannelFuture.addListener((ChannelFutureListener) future -> {
+        webSocketChannelFuture.addListener(future -> {
             Throwable cause = future.cause();
             if (!future.isSuccess() && cause != null) {
-                context.setReturnValues(createWebSocketConnectorError(context, future.cause().getMessage()));
+                context.setReturnValues(createWebSocketConnectorError(context, cause));
             } else {
                 context.setReturnValues();
             }
             callback.notifySuccess();
         });
     }
 
-    public static BStruct createWebSocketConnectorError(Context context, String errorMsg) {
-        return BLangConnectorSPIUtil
-                .createBStruct(context, HttpConstants.PROTOCOL_PACKAGE_HTTP,
-                               WebSocketConstants.WEBSOCKET_CONNECTOR_ERROR, errorMsg,
-                               new BValue[]{});
+    public static BStruct createWebSocketConnectorError(Context context, Throwable throwable) {
+        return HttpUtil.getHttpConnectorError(context, throwable);
```

Comment:
In my PR https://github.com/ballerina-platform/ballerina-lang/pull/8549/files I have fixed this in all places by removing this method and using instead this `BLangVMErrors.createError(context, cause.getMessage())` in all places. Once that PR is merged the issue with errors would be gone so there's no need of this change in this PR

Reply:
Revert back the changes